apps/rush-lib/src/data/EnvironmentConfiguration.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+const ENV_VALUE_PREFIX: string = 'rush_';
+
+export enum EnvironmentValue {
+  TempDirectoryOverride = 'rush_tempDir'
+}
+
+const NORMALIZED_ENV_VALUE_NAMES: { [lowercaseValue: string]: EnvironmentValue } = {};
+for (const envValue in EnvironmentValue) {
+  if (EnvironmentValue.hasOwnProperty(envValue)) {
+    const enumValue: string = EnvironmentValue[envValue];
+    NORMALIZED_ENV_VALUE_NAMES[enumValue.toLowerCase()] = enumValue as EnvironmentValue;
```

Comment:
>toLowerCase [](start = 41, length = 11)

When normalizing case, we always use toUpperCase() because (1) it handles Turkish "I" correctly and (2) it makes it obvious in the debugger that the case has been normalized.

However, if you take my suggestion above, then the names should be already normalized.  (Don't attempt to normalize the OS environment variables, since for some shells they are expected to be case-sensitive)